### PR TITLE
[params] Let user specify city centre and gradient weights

### PR DESCRIPTION
**Pull request by NicEastvillage**
Add the following parameters:
```
    --centre.pos args           The coordinates for the city's centre, e.g. "300,500" or "auto" [default: auto]
    --centre.pop-weight F       The increase in population near the city center [default: 0.8]
    --centre.work-weight F      The increase in work places near the city center [default: 0.1]
```

### CODE REVIEW

**Review by falkecarlsen — APPROVED** [review]

File: randomActivityGen.py [hunk 1]
```diff
@@ -10,6 +11,9 @@
     -o, --output-file FILE      Write modified statistics to FILE
 
 Other Options:
+    --centre.pos args           The coordinates for the city's centre, e.g. "300,500" or "auto" [default: auto]
+    --centre.pop-weight F       The increase in population near the city center [default: 0.8]
```
Comment (MathiasMehl) [inline randomActivityGen.py:15]: Intuitively I would think of 0.8 as being a 20% decrease, not an 80% increase, similar for centre.work-weight. But works as it is defined as an increase.

File: randomActivityGen.py [hunk 2]
```diff
@@ -229,21 +233,23 @@ def main():
...
+    apply_network_noise(net, stats, centre, float(args["--centre.pop-weight"]), float(args["--centre.work-weight"]))
 
     setup_city_gates(net, stats, int(args["--gates.count"]))
 
     if args["--schools.count"] == "auto":
-        setup_schools(net, stats, None)
+        setup_schools(net, stats, None, centre)
     else:
-        setup_schools(net, stats, int(args["--schools.count"]))
+        setup_schools(net, stats, int(args["--schools.count"]), centre)
```
Comment (MathiasMehl) [inline randomActivityGen.py:256]: Why is centre being added to school functions?

File: randomActivityGen.py (hunk 1 above)
Comment (NicEastvillage) [reply randomActivityGen.py:15]: Yeah, I see what you mean. But it is a weight for the "added" gradient, so it kinda works like an increase.

File: randomActivityGen.py (hunk 2 above)
Comment (NicEastvillage) [reply randomActivityGen.py:256]: Because the placement of schools relies on the population noise which relies on the location of the city centre.

**Review by MathiasMehl — APPROVED** [review]

**Pull request merged**